shared/src/main/scala/pl/metastack/metarx/Buffer.scala

```diff
@@ -591,6 +591,18 @@ trait PollBuffer[T]
       get.foldLeft(acc)(f)
     }.distinct
 
+  def reduce[U >: T](op: (U, U) ⇒ U): ReadChannel[U] =
+    changes.map { _ =>
+      get.reduce(op)
+    }.distinct
+
+  def mkString(sep: String): ReadChannel[String] = {
```

Comment:
Sure, I have programmed too much Java (and I think the original implementation was doing it like that).